<?xml version = "1.0" ?>
<robot name="arm_mesh">
	<link name = "base">
			<visual>
				<origin rpy="0 0 0" xyz = "0 0 0"/>
				<geometry>
               		<cylinder radius="0.05" length="0.035"/>
            	</geometry>
        	</visual>

            <collision>
                <origin rpy="0 0 0" xyz = "0 0 0"/>
                <geometry>
                    <cylinder radius="0.05" length="0.035"/>
                </geometry>
            </collision>

            <inertial>
                <mass value = "0.170"/>
                <origin rpy="0 0 0" xyz = "0 0 0"/>
                <inertia ixx="8.7e-05" ixy="0.0" ixz="0.0" iyy="8.7e-05" iyz="0.0" izz="0.00015"/>
            </inertial>

    </link>


    <joint name = "base_link1" type = "revolute">
			<axis xyz= "0 0 1"/>
			<limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5" />
        	<origin rpy="0 0 0" xyz="0 0 0.0175"/>
        	<parent link="base"/>
        	<child link="link1"/>
    </joint>

     <transmission name="trans_base_link1">
      <type>transmission_interface/SimpleTransmission</type>
      <joint name="base_link1">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      </joint>
      <actuator name="motor_base_link1">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        <mechanicalReduction>1</mechanicalReduction>
      </actuator>
    </transmission>

    <link name = "link1">
			<visual>
				<origin rpy="0 0 0" xyz = "0 0 0.015"/>
				<geometry>
               		<cylinder radius="0.035" length="0.03"/>
            	</geometry>
        	</visual>

            <collision>
                <origin rpy="0 0 0" xyz = "0 0 0.015"/>
                <geometry>
                    <cylinder radius="0.035" length="0.03"/>
                </geometry>
            </collision>

            <inertial>
                <mass value = "0.075"/>
                <origin rpy="0 0 0" xyz = "0 0 0.0325 "/>
                <inertia ixx="0.037" ixy="0.000274" ixz="-0.000001395" iyy="0.0245" iyz="0.000007207" izz="0.01526"/>
            </inertial>
    </link>

    <joint name = "link1_link2" type = "revolute">
			<axis xyz= "0 1 0"/>
			<limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5" />
        	<origin rpy="0 0 0" xyz="0 0 0.042"/>
        	<parent link="link1"/>
        	<child link="link2"/>
        </joint>

    <transmission name="trans_link1_link2">
      <type>transmission_interface/SimpleTransmission</type>
      <joint name="link1_link2">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      </joint>
      <actuator name="motor_link1_link2">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        <mechanicalReduction>1</mechanicalReduction>
      </actuator>
    </transmission>

    <link name = "link2">
			<visual>
				<origin rpy="0 0 0" xyz = "0 -0.02 0.05"/>
				<geometry>
               		<mesh filename="package://robotic_arm_pkg/meshes/link2.stl" scale = "0.001 0.001 0.001"/>
            	</geometry>
        	</visual>
            <collision>
                <origin rpy="0 0 0" xyz = "0 0 0.05"/>
                <geometry>
                    <box size ="0.025 0.054 0.12"/>
                </geometry>
            </collision>

            <inertial>
                <mass value = "0.018"/>
                <origin rpy="0 0 0" xyz = "0 0 -0.005 "/>
                <inertia ixx="0.037" ixy="0.000274" ixz="-0.000001395" iyy="0.0245" iyz="0.000007207" izz="0.01526"/>
            </inertial>
    </link>


    <joint name = "link2_link3" type = "revolute">
			<axis xyz= "0 1 0"/>
			<limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5" />
        	<origin rpy="0 0 0" xyz="0 0 0.095"/>
        	<parent link="link2"/>
        	<child link="link3"/>
        </joint>

    <transmission name="trans_link2_link3">
      <type>transmission_interface/SimpleTransmission</type>
      <joint name="link2_link3">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      </joint>
      <actuator name="motor_link2_link3">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        <mechanicalReduction>1</mechanicalReduction>
      </actuator>
    </transmission>

    <link name = "link3">
			<visual>
				<origin rpy="0 0 0" xyz = "-0.04 -0.045 0.03"/>
				<geometry>
               		<mesh filename="package://robotic_arm_pkg/meshes/link3.stl" scale = "0.001 0.001 0.001"/>
            	</geometry>
        	</visual>

            <collision>
                <origin rpy="0 0 0" xyz = "0 0 0.045"/>
                <geometry>
                    <box size="0.03 0.04 0.14"/>
                </geometry>
            </collision>

            <inertial>
                <mass value = "0.130"/>
                <origin rpy="0 0 0" xyz = "0 0 -0.01 "/>
                <inertia ixx="0.037" ixy="0.000274" ixz="-0.000001395" iyy="0.0245" iyz="0.000007207" izz="0.01526"/>
            </inertial>
    </link>

    <joint name = "link3_link4" type = "revolute">
			<axis xyz= "0 1 0"/>
			<limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5" />
        	<origin rpy="0 0 0" xyz="0 0 0.09"/>
        	<parent link="link3"/>
        	<child link="link4"/>
        </joint>

    <transmission name="trans_link3_link4">
      <type>transmission_interface/SimpleTransmission</type>
      <joint name="link3_link4">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      </joint>
      <actuator name="motor_link3_link4">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        <mechanicalReduction>1</mechanicalReduction>
      </actuator>
    </transmission>

    <link name = "link4">
			<visual>
				<origin rpy="0 -1.57 1.57" xyz = "0.017 -0.0135 0.033"/>
				<geometry>
               		<mesh filename="package://robotic_arm_pkg/meshes/link4.stl" scale = "0.001 0.001 0.001"/>
            	</geometry>
        	</visual>

            <collision>
                <origin rpy="0 -1.57 1.57" xyz = "0.017 -0.0135 0.033"/>
                <geometry>
                    <mesh filename="package://robotic_arm_pkg/meshes/link4.stl" scale = "0.001 0.001 0.001"/>
                </geometry>
            </collision>

            <inertial>
                <mass value = "0.025"/>
                <origin rpy="0 0 0" xyz = "0 0 0 "/>
                <inertia ixx="0.00485" ixy="-0.000197" ixz="-0.00017" iyy="0.0084" iyz="0.0" izz="0.0042"/>
            </inertial>
    </link>

    <joint name = "link4_gripper_base" type = "revolute">
			<axis xyz= "0 0 1"/>
			<limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5" />
        	<origin rpy="0 0 0" xyz="-0.01 0 0.06"/>
        	<parent link="link4"/>
        	<child link="gripper_base"/>
        </joint>

    <transmission name="trans_link4_gripper_base">
      <type>transmission_interface/SimpleTransmission</type>
      <joint name="link4_gripper_base">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      </joint>
      <actuator name="motor_link4_gripper_base">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        <mechanicalReduction>1</mechanicalReduction>
      </actuator>
    </transmission>

   <link name = "gripper_base">
			<visual>
				<origin rpy="0 0 3.14" xyz = "0.1 0.095 0"/>
				<geometry>
               		<mesh filename="package://robotic_arm_pkg/meshes/gripper_base.stl" scale = "0.001 0.001 0.001"/>
            	</geometry>
        	</visual>

            <collision>
                <origin rpy="0 0 3.14" xyz = "0.1 0.095 0"/>
                <geometry>
                    <mesh filename="package://robotic_arm_pkg/meshes/gripper_base.stl" scale = "0.001 0.001 0.001"/>
                </geometry>
            </collision>

            <inertial>
                <mass value = "0.020"/>
                <origin rpy="0 0 0" xyz = "0 0 0 "/>
                <inertia ixx="0.00156" ixy="0.0000274" ixz="-0.00001845" iyy="0.0016" iyz="0.000032" izz="0.0027"/>
            </inertial>
    </link>
    

 <joint name = "gripper_base_claw_L" type = "revolute">
			<axis xyz= "1 0 0"/>
			<limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5" />
        	<origin rpy="0 0 0" xyz="0 -0.0125 0.01"/>
        	<parent link="gripper_base"/>
        	<child link="claw_L"/>
        </joint>

    <link name = "claw_L">
			<visual>
				<origin rpy="-1.57 0 0" xyz = "-0.150 -0.01 0.125"/>
				<geometry>
               		<mesh filename="package://robotic_arm_pkg/meshes/gripper_claw_L.stl" scale = "0.001 0.001 0.001"/>
            	</geometry>
        	</visual>
            <collision>
                <origin rpy="-1.57 0 0" xyz = "-0.150 -0.01 0.125"/>
                <geometry>
                    <mesh filename="package://robotic_arm_pkg/meshes/gripper_claw_L.stl" scale = "0.001 0.001 0.001"/>
                </geometry>
            </collision>

            <inertial>
                <mass value = "0.007"/>
                <origin rpy="0 0 0" xyz = "0 0 0 "/>
                <inertia ixx="0.037" ixy="0.000274" ixz="-0.000001395" iyy="0.0245" iyz="0.000007207" izz="0.01526"/>
            </inertial>
    </link>

    <joint name = "gripper_base_claw_R" type = "revolute">
			<axis xyz= "-1 0 0"/>
			<limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5" />
        	<origin rpy="0 0 0" xyz="0 0.0035 0.012"/>
        	<parent link="gripper_base"/>
        	<child link="claw_R"/>
        </joint>

    <link name = "claw_R">
			<visual>
				<origin rpy="-1.57 0 3.14" xyz = "0.147 0.01 0.123"/>
				<geometry>
               		<mesh filename="package://robotic_arm_pkg/meshes/gripper_claw_R.stl" scale = "0.001 0.001 0.001"/>
            	</geometry>
        	</visual>
            <collision>
                <origin rpy="-1.57 0 3.14" xyz = "0.147 0.01 0.123"/>
                <geometry>
                    <mesh filename="package://robotic_arm_pkg/meshes/gripper_claw_R.stl" scale = "0.001 0.001 0.001"/>
                </geometry>
            </collision>

            <inertial>
                <mass value = "0.007"/>
               <origin rpy="0 0 0" xyz = "0 0 0 "/>
                <inertia ixx="0.037" ixy="0.000274" ixz="-0.000001395" iyy="0.0245" iyz="0.000007207" izz="0.01526"/>
            </inertial>
    </link>


 <gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
    <robotNamespace>/arm_mesh</robotNamespace>
    <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
    <legacyModeNS>true</legacyModeNS>
    </plugin>
  </gazebo>


   

</robot>